### ISSUE TITLE
Config out of date error message does not go away regardless of updating / copying file over.

### ISSUE THREAD

**Issue by dubwoc** [ampache/ampache#2533]
## Description
When logging into Ampache I get the following at the top and I cannot make it go away:
```
Your Ampache config file is out of date!
Download a new config file to manually update | Write a new config file directly to disk
```

### Describe the bug
When I click on Download new config and copy it over, there are no differences, when I click write config, it appears to write fine but there are no changes and the message does not go away.

I know it is writing fine because I have debugging enabled and I played around with the permissions to see it fail properly in the logs and then I changed the permissions back to see it succeed silently.

### To reproduce

1. Log in.
2. Observe the message at the top of the main page saying my config is out of date. 
3. Try downloading the config or clicking write config to fix the problem.
4. Continue to see error about config being out of date. 

### Expected behavior
After clicking write config i'd expect this to go away. 

## Environment

* Ampache version: 4.2.2
* Web server + version: Apache 2.4.38-3+deb10u4
* Server operating system: Debian 10 Buster
* Client operating system: Debian, but really any OS I browse to the web app on. 

### Client type

* Web

## Settings
Config below without commented out lines:
```
config_version = 40
web_path = ""
database_hostname = "localhost"
database_port = ""
database_name = "ampache"
database_username = "ampache"
database_password = "XXXXXX"
secret_key = "XXXXXX"
session_length = "3600"
stream_length = "7200"
remember_length = "604800"
session_name = "ampache"
session_cookielife = "0"
session_cookiesecure = "0"
auth_methods = "mysql"
access_control = "true"
require_session = "true"
require_localnet_session = "true"
getid3_tag_order = "id3v2,id3v1,vorbiscomment,quicktime,matroska,ape,asf,avi,mpeg,riff"
metadata_order = "getID3,filename"
metadata_order_video = "filename,getID3"
deferred_ext_metadata = "true"
additional_genre_delimiters = "[/]{2}|[/|\|\|,|;]"
catalog_file_pattern = "mp3|mpc|m4p|m4a|aac|ogg|oga|wav|aif|aiff|rm|wma|asf|flac|opus|spx|ra|ape|shn|wv"
catalog_video_pattern = "avi|mpg|mpeg|flv|m4v|mp4|webm|mkv|wmv|ogv|mov|divx|m2ts"
catalog_playlist_pattern = "m3u|m3u8|pls|asx|xspf"
catalog_prefix_pattern = "The|An|A|Die|Das|Ein|Eine|Les|Le|La"
use_auth = "true"
default_auth_level = "guest"
ratings = "true"
userflags = "true"
directplay = "true"
sociable = "true"
licensing = "false"
max_upload_size = "10048576"
art_order = "folder,db,tags,musicbrainz,lastfm,google"
show_similar = "true"
show_concerts = "true"
lastfm_api_key = "f8214f476b8607c1977fa7d2d35e3444"
lastfm_api_secret = "b78fe34b1673bd6fe77ccea26d9de07e"
wanted = "true"
wanted_types = "album,official"
channel = "true"
live_stream = "true"
podcast = "true"
refresh_limit = "60"
show_footer_statistics = "true"
debug = "true"
debug_level = "5"
log_path = "/var/log/ampache"
log_filename = "%name.%Y%m%d.log"
site_charset = "UTF-8"
allow_public_registration = "false"
cookie_disclaimer = "false"
registration_display_fields = "fullname,website"
registration_mandatory_fields = "fullname"
encode_target = "mp3"
transcode_player_customize = "true"
transcode_cmd = "avconv"
transcode_input = "-i %FILE%"
encode_args_mp3 = "-vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1"
encode_args_ogg = "-vn -b:a %BITRATE%K -c:a libvorbis -f ogg pipe:1"
encode_args_opus = "-vn -b:a %BITRATE%K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1"
encode_args_m4a = "-vn -b:a %BITRATE%K -c:a libfdk_aac -f adts pipe:1"
encode_args_wav = "-vn -b:a %BITRATE%K -c:a pcm_s16le -f wav pipe:1"
encode_args_flv = "-b:a %BITRATE%K -ar 44100 -ac 2 -v 0 -f flv -c:v libx264 -preset superfast -threads 0 pipe:1"
encode_args_webm = "-q %QUALITY% -f webm -c:v libvpx -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_args_ts = "-q %QUALITY% -s %RESOLUTION% -f mpegts -c:v libx264 -c:a libmp3lame -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_args_ogv = "-codec:v libtheora -qscale:v 7 -codec:a libvorbis -qscale:a 5 -f ogg pipe:1"
encode_get_image = "-ss %TIME% -f image2 -vframes 1 pipe:1"
encode_srt = "-vf \"subtitles='%SRTFILE%'\""
encode_ss_frame = "-ss %TIME%"
encode_ss_duration = "-t %DURATION%"
force_ssl = "true"
common_abbr = "divx,xvid,dvdrip,hdtv,lol,axxo,repack,xor,pdtv,real,vtv,caph,2hd,proper,fqm,uncut,topaz,tvt,notv,fpn,fov,orenji,0tv,omicron,dsr,ws,sys,crimson,wat,hiqt,internal,brrip,boheme,vost,vostfr,fastsub,addiction,x264,LOL,720p,1080p,YIFY,evolve,fihtv,first,bokutox,bluray,tvboom,info"
```
## Logs

Nothing in the debug logs looks specific to this problem.

**Comment by wagnered** [comment]
Confg_version 40 was released in  Ampache 4.03 back in Nov 2019.  The config_version for 4.2.2 is 45.  There might have been db changes between the two: I suggest doing a reinstall of 4.2.2 or the develop branch.

Are you sure it's the latest release version and not the develop branch?  The reason I ask is that a release, by default is fixed and should never get that message.

**Comment by dubwoc** [comment]
I am positive. the container is created with this line: 

ADD https://github.com/ampache/ampache/archive/4.2.2.tar.gz /opt/archive.tar.gz


Is this of any help as far as whether or not I am missing a db change? 

> MariaDB [ampache]> select * FROM update_info;
> +--------------------+--------+
> | key                | value  |
> +--------------------+--------+
> | db_version         | 400011 |
> | Plugin_Last.FM     | 000005 |
> | Plugin_MusicBrainz | 000001 |
> | Plugin_Discogs     | 000001 |
> | Plugin_TheAudioDb  | 000002 |
> | Plugin_Omdb        | 000001 |
> | Plugin_LyricWiki   | 000001 |
> +--------------------+--------+
> 7 rows in set (0.186 sec)
> 

When you say "reinstall" I assume you are saying not starting from scratch but making sure I am running a version of 4.2.2 in the container I am running. If that is the case, I have done that and I still have this issue.

I've updated a number of times between November 2019 and now so I am not sure why my config version is stated as so low. 

Thank you for helping / suggesting ways forward.

**Comment by dubwoc** [comment]
I think I found my problem. At some point a while ago, I think in between the 3.8.2 and 3.9.x days I got an error regarding the ampache.cfg.php.dist not being present so I copied one into my config directory and that is why when I write the new config via the error message describe above it gives me the older one as it is using an out of date ampache.cfg.php.dist file. 

I put an updated dist file in place and generated the latest config, checked the updates, and that error is gone. I appear to be in good shape now. 

I think I can fix this from my container set up by changing what I mount into the container to be the config and not a whole directory that has other files, including .dist files that should be updated as part of the distribution.

Thanks for the help again I think we can close this unless you would like me to describe anything further.
